config/management/operational/src/network_checker.rs

```diff
@@ -80,6 +83,73 @@ impl CheckEndpoint {
     }
 }
 
+#[derive(Debug, StructOpt)]
+pub struct CheckValidatorSetEndpoints {
+    /// JSON-RPC Endpoint (e.g. http://localhost:8080)
+    #[structopt(long, required_unless = "config")]
+    json_server: String,
+    /// Specifies whether or not to evaluate validators or fullnodes
+    #[structopt(long)]
+    role: RoleType,
+    /// The expected on-chain key, only required for validator checks
+    #[structopt(long, required_if("role", "validator"), parse(try_from_str = parse_hex))]
+    key: Option<Key>,
+    /// The expected on-chain key version, only required for validator checks
+    #[structopt(long, required_if("role", "validator"))]
+    version: Option<KeyVersion>,
+}
+
+fn parse_hex(src: &str) -> Result<Key, Error> {
+    let potential_err_msg = format!("Not a valid encryption key: {}", src);
+    let value_slice = match hex::decode(src.trim()) {
+        Ok(value_slice) => {
+            if value_slice.len() != KEY_LEN {
+                Err(Error::CommandArgumentError(potential_err_msg))
+            } else {
+                Ok(value_slice)
+            }
+        }
+        Err(_) => Err(Error::CommandArgumentError(potential_err_msg)),
+    }?;
+
+    let mut value = [0; KEY_LEN];
+    value.copy_from_slice(&value_slice);
+    Ok(value)
+}
+
+impl CheckValidatorSetEndpoints {
+    pub fn execute(self) -> Result<String, Error> {
+        let encryptor = match self.role {
+            RoleType::FullNode => Encryptor::for_testing(),
```

Comment:
nit: it isn't great that we use "Encryptor::for_testing()" in non-test code.. The comment above "for_testing" also doesn't bode well (e.g., "/// This generates an Encryptor for use in default / testing scenarios where (proper) encryption is not necessary."). Perhaps we could fix/rename these?

Moreover, I guess we don't want to use Encryptor::new because of the backend requirement? Can we not support both a backend or a command line key (i.e., use "for_testing" if passing the key in, otherwise "new" with a backend?)

Reply:
you guys suck.

basically I just wanted a Encryptor with in memory storage lol... I'll add the extra couple lines of code...